spring-boot/autoconfigure/src/main/java/org/togglz/spring/boot/autoconfigure/TogglzAutoConfiguration.java

```diff
@@ -0,0 +1,251 @@
+            if (stateRepositories.size() == 1) {
+                stateRepository = stateRepositories.get(0);
+            } else if (stateRepositories.size() > 1) {
+                stateRepository = new CompositeStateRepository(
+                        stateRepositories.toArray(new StateRepository[stateRepositories.size()]));
+            }
+            // If caching is enabled wrap state repository in caching state
+            // repository.
+            // Note that we explicitly check if the state repository is not
+            // already a caching state repository,
+            // as the auto configuration of the state repository already creates
+            // a caching state repository if needed.
+            // The below wrapping only occurs if the user provided the state
+            // repository manually and caching is enabled.
+            if (properties.getCache().isEnabled() && !(stateRepository instanceof CachingStateRepository)) {
+                stateRepository = new CachingStateRepository(stateRepository, properties.getCache().getTimeToLive(),
+                        properties.getCache().getTimeUnit());
+            }
+            FeatureManagerBuilder featureManagerBuilder = new FeatureManagerBuilder();
+            String name = properties.getFeatureManagerName();
+            if (name != null && name.length() > 0) {
+                featureManagerBuilder.name(name);
+            }
+            featureManagerBuilder.featureProvider(featureProvider).stateRepository(stateRepository)
+                    .userProvider(userProvider).activationStrategyProvider(activationStrategyProvider).build();
+            FeatureManager manager = featureManagerBuilder.build();
+            return manager;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingBean(ActivationStrategyProvider.class)
+    protected static class ActivationStrategyProviderConfiguration {
+
+        @Autowired(required = false)
+        private List<ActivationStrategy> activationStrategies;
+
+        @Bean
+        public ActivationStrategyProvider activationStrategyProvider() {
+            DefaultActivationStrategyProvider provider = new DefaultActivationStrategyProvider();
+            if (activationStrategies != null && activationStrategies.size() > 0) {
+                provider.addActivationStrategies(activationStrategies);
+            }
+            return provider;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingBean(StateRepository.class)
+    protected static class StateRepositoryConfiguration {
+
+        @Autowired
+        private ResourceLoader resourceLoader = new DefaultResourceLoader();
+
+        @Autowired
+        private TogglzProperties properties;
+
+        @Bean
+        public StateRepository stateRepository() throws IOException {
+            StateRepository stateRepository;
+            String featuresFile = properties.getFeaturesFile();
+            if (featuresFile != null) {
```

Comment:
In order to keep number of nested blocks lower you could go for:

```
if (featuresFile == null) { 
 Map<String, FeatureSpec> features = properties.getFeatures();
                stateRepository = new InMemoryStateRepository();
                for (String name : features.keySet()) {
                    stateRepository.setFeatureState(features.get(name).state(name));
                }
} else {
....
}
```

Reply:
I didn't get that. We would still need to handle the `featuresFile != null` in the `else` block,